core/src/main/java/org/apache/carbondata/core/metadata/ValueEncoderMeta.java

```diff
@@ -32,23 +39,37 @@
   /**
    * maxValue
    */
-  private Object maxValue;
+  protected Object maxValue;
```

Comment:
I feel it is not safe to change even method name or access specifier, just extend this class or wrap it into another class to add more variables or do any changes.